src/interactable_music_plant_german_flute.c

```diff
@@ -0,0 +1,384 @@
+#include "global.h"
+#include "m4a.h"
+#include "task.h"
+#include "trig.h"
+
+#include "interactable.h"
+#include "interactable_music_plant_note_particle.h"
+
+#include "constants/move_states.h"
+#include "constants/songs.h"
+
+typedef struct {
+    /* 0x00 */ s32 posX;
+    /* 0x04 */ s32 posY;
+    /* 0x08 */ s16 unk8;
+    /* 0x0A */ s16 unkA;
+    /* 0x0C */ u16 kind;
+    /* 0x0E */ u16 timer;
+    /* 0x10 */ Interactable *ia;
+    /* 0x14 */ u8 spriteX;
+    /* 0x15 */ u8 spriteY;
+    /* 0x16 */ u8 unk16;
+    /* 0x17 */ u8 unk17;
+} Sprite_GermanFlute; /* size: 0x18 */
+
+#define NUM_GERMAN_FLUTE_KINDS 4
+
+typedef void (*SomeFunc)(s32, s32, s32);
+
+extern void sub_801EC3C(s32, s32, s32);
+extern s32 sub_801F100(s32, s32, s32, s32, SomeFunc);
+
+extern void sub_80218E4(Player *);
+extern void sub_8023B5C(Player *, u32);
+
+static void sub_8076928(void);
+static void sub_80769E0(void);
+static void Task_8076A6C(void);
+static void sub_8076B84(Sprite_GermanFlute *);
+static bool32 sub_8076BE4(Sprite_GermanFlute *);
+static void sub_8076C70(Sprite_GermanFlute *);
+static void sub_8076C88(Sprite_GermanFlute UNUSED *);
+static void sub_8076CC0(Sprite_GermanFlute UNUSED *);
+static void sub_8076CF4(Sprite_GermanFlute UNUSED *);
+static void sub_8076D08(Sprite_GermanFlute UNUSED *);
+static void Task_Interactable_MusicPlant_GermanFlute(void);
+static void Task_8076DE8(void);
+static void TaskDestructor_Interactable_MusicPlant_GermanFlute(struct Task *);
+static void sub_8076E3C(Sprite_GermanFlute *);
+static bool32 sub_8076EAC(Sprite_GermanFlute *);
+static void sub_8076EF4(Sprite_GermanFlute *);
+static void sub_8076C58(Sprite_GermanFlute *);
+
+static const s16 sFluteUpdraft[NUM_GERMAN_FLUTE_KINDS] = {
+    Q_8_8(7),
+    Q_8_8(8),
+    Q_8_8(9),
+    Q_8_8(10),
+};
+
+static const u16 sFluteSfx[NUM_GERMAN_FLUTE_KINDS] = {
+    SE_MUSIC_PLANT_FLUTE_1,
+    SE_MUSIC_PLANT_FLUTE_2,
+    SE_MUSIC_PLANT_FLUTE_3,
+    SE_MUSIC_PLANT_FLUTE_4,
+};
+
+static void sub_8076928(void)
+{
+    Sprite_GermanFlute *flute = TaskGetStructPtr(gCurTask);
+
+    if (gPlayer.moveState & MOVESTATE_DEAD) {
+        sub_8076CF4(flute);
+    } else if (gPlayer.unk2C == 0x78) {
+        sub_8076D08(flute);
+    } else {
+        s32 posX, posY;
+        posX = Q_24_8(flute->posX);
+        posY = Q_24_8(flute->posY) + Q_24_8(24);
+
+        if (gPlayer.x != posX) {
+            if (gPlayer.x > posX) {
+                gPlayer.x -= Q_24_8(0.5);
+
+                if (gPlayer.x < posX) {
+                    gPlayer.x = posX;
+                }
+
+            } else {
+                gPlayer.x += Q_24_8(0.5);
+
+                if (gPlayer.x > posX) {
+                    gPlayer.x = posX;
+                }
+            }
+        }
+
+        if ((gPlayer.x - posX) >= 0) {
+            if ((gPlayer.x - posX) <= Q_24_8(8)) {
+                goto lab;
```

Comment:
You can apply the abs fix to this now :P

Reply:
Oh yeah, sure! :D